src/api/cvms.ts

```diff
@@ -0,0 +1,333 @@
+ * @returns Success status
+ */
+export async function stopCvm(appId: string): Promise<boolean> {
+  try {
+    await apiClient.post(API_ENDPOINTS.CVM_STOP(appId));
+    return true;
+  } catch (error) {
+    throw new Error(`Failed to stop CVM: ${error instanceof Error ? error.message : String(error)}`);
+  }
+}
+
+/**
+ * Restart a CVM
+ * @param appId App ID
+ * @returns Success status
+ */
+export async function restartCvm(appId: string): Promise<boolean> {
+  try {
+    await apiClient.post(API_ENDPOINTS.CVM_RESTART(appId));
+    return true;
+  } catch (error) {
+    throw new Error(`Failed to restart CVM: ${error instanceof Error ? error.message : String(error)}`);
+  }
+}
+
+/**
+ * Upgrade a CVM
+ * @param appId App ID
+ * @param vmConfig VM configuration
+ * @returns Upgrade response
+ */
+export async function upgradeCvm(appId: string, vmConfig: any): Promise<UpgradeCvmResponse> {
+  try {
+    const response = await apiClient.put<UpgradeCvmResponse>(API_ENDPOINTS.CVM_UPGRADE(appId), vmConfig);
+    return upgradeCvmResponseSchema.parse(response);
+  } catch (error) {
+    throw new Error(`Failed to upgrade CVM: ${error instanceof Error ? error.message : String(error)}`);
+  }
+}
+
+/**
+ * Helper function to convert hex string to Uint8Array
+ */
+function hexToUint8Array(hex: string): Uint8Array {
+  return new Uint8Array(hex.match(/.{1,2}/g)!.map(byte => parseInt(byte, 16)));
+}
+
+/**
+ * Helper function to convert Uint8Array to hex string
+ */
+function uint8ArrayToHex(bytes: Uint8Array): string {
+  return Array.from(bytes)
+    .map(b => b.toString(16).padStart(2, '0'))
+    .join('');
+}
+
+/**
+ * Encrypt environment variables for CVM
+ * @param secrets Environment variables
+ * @param pubkey Public key
+ * @returns Encrypted environment variables
+ */
+export async function encryptSecrets(secrets: Env[], pubkey: string): Promise<string> {
+  const envsJson = JSON.stringify({ env: secrets });
+
+  // Generate private key and derive public key
+  const privateKey = x25519.utils.randomPrivateKey();
+  const publicKey = x25519.getPublicKey(privateKey);
+
+  // Generate shared key
+  const remotePubkey = hexToUint8Array(pubkey);
+  const shared = x25519.getSharedSecret(privateKey, remotePubkey);
+
+  // Import shared key for AES-GCM
+  const importedShared = await crypto.subtle.importKey(
+    'raw',
+    shared,
+    { name: 'AES-GCM', length: 256 },
+    true,
+    ['encrypt'],
+  );
+
+  // Encrypt the data
+  const iv = crypto.getRandomValues(new Uint8Array(12));
+  const encrypted = await crypto.subtle.encrypt(
+    { name: 'AES-GCM', iv },
+    importedShared,
+    new TextEncoder().encode(envsJson),
+  );
+
+  // Combine all components
+  const result = new Uint8Array(
+    publicKey.length + iv.length + encrypted.byteLength,
+  );
+
+  result.set(publicKey);
+  result.set(iv, publicKey.length);
+  result.set(new Uint8Array(encrypted), publicKey.length + iv.length);
+
+  return uint8ArrayToHex(result);
+}
```

Comment:
Just a reminder, we can use the one from dstack-sdk that I added last week.
https://www.npmjs.com/package/@phala/dstack-sdk#environment-variables-encryption